exec/java-exec/src/main/java/org/apache/drill/exec/vector/complex/fn/JsonReader.java

```diff
@@ -59,6 +59,12 @@
   private final boolean readNumbersAsDouble;
 
   /**
+   * Collection for tracking empty array writers during reading
+   * and storing them for initializing empty arrays
+   */
+  private final Set<ListWriter> emptyArrayWritersSet = Sets.newHashSet();
```

Comment:
But it should be mentioned that this list may contain many duplicates, as every ListWriter from every empty array field will be added into it. 
To avoid secondary call field.integer() on the same ListWriter object, which was added many times into the list, in the method JsonReader.ensureAtLeastOneField() we check valueCapacity of every ListWriter. 
Also LinkedHashSet may be used there, but in the case of many empty array fields it may cause additional extra cost.